### PR TITLE
Fetch Config from URL

### LINKED ISSUE #113: Fetch Config from URL
Cowswap will likely keep the config file in a github repo. We would like to allow URL for config file that can be fetched form the github repo.

### PR DESCRIPTION
**Pull request by mooster531**
Closes #113

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -163,8 +204,12 @@ def load(cls, file_path: Path | str = "config.yaml") -> RuntimeConfig:
             ValueError: If the configuration contains invalid database types
 
         """
-        with open(file_path, encoding="utf-8") as _handle:
-            data = cls.read_yaml(_handle)
+        # TODO maybe support other schemes - what if it lives on an FTP server? :^)
+        #  alternatively, enforce scheme - files would be passed as "file:///path/to/config.yaml"
+        if re.match(r"^https?://.*", str(file_path)):
```
Comment (bh2smith) [inline src/config.py]: Doesn't python have some library like parse url or is url so we can try parse url or assume path. Might be more robust than regex.

**Review by bh2smith — APPROVED** [review]
Looks good although I think we should change the regex for some existing url parser. How do our mocked tests ensure that the fetching actually works? Can we use a real example where we reference the config that lives in this repo by URL?

File: src/config.py (hunk 1 above)
Comment (mooster531) [reply src/config.py]: Fair point, it's probably a better idea to use something in `urllib` that either returns a URL or None, I'll fix it.

Comment (mooster531) [reply src/config.py]: > How do our mocked tests ensure that the fetching actually works

Well the fetching is performed by `aiohttp` and I'd rather not test external code, but I agree that an end-to-end test would be beneficial. I'll fix one up.

**Comment by mooster531** [comment]
Addressed the remarks in previous review round, please feel free to take another look.

File: tests/e2e_test.py [hunk 2]
```diff
@@ -207,3 +208,17 @@ async def test_dune_to_local_job_run(self, mock_env, mock_dune_client):
             await conf.jobs[0].run()
 
         self.assertIn("No Query results found! Skipping write", logs.output[0])
+
+    @patch("src.config.load_dotenv")
+    @patch.dict(os.environ, {"DUNE_API_KEY": "test_key", "DB_URL": DB_URL})
+    @skipIf(not os.getenv("CI"), "this test only runs in CI")
```
Comment (bh2smith) [inline tests/e2e_test.py:214]: Neat! Learned something!

**Review by bh2smith — APPROVED** [review]
Way cool! Nice job.

File: tests/e2e_test.py (hunk 2 above)
Comment (mooster531) [reply tests/e2e_test.py:214]: So did I - what env vars are set in Github Actions!
Reference: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables

**Pull request merged**